spd/losses.py

```diff
@@ -138,12 +138,10 @@ def calc_masked_recon_layerwise_loss(
     assert loss_type in ["mse", "kl"], f"Invalid loss type: {loss_type}"
     total_loss = torch.tensor(0.0, device=device)
     for mask_info in masks:
-        for component_name in model.components:
-            modified_out = model(
-                batch,
-                mode="components",
-                masks={component_name: mask_info[component_name]},
-            )
+        for comp_name, mask in mask_info.items():
+            # TODO: Write a test showing that passing a mask for a ComponentOrModule which has
+            # both components and identity components works as expected.
```

Comment:
Flagging this. Should be done before merging. Should be some other tests of this too.